src/test/java/com/sigma/cinematicketpro/controller/MovieControllerTest.java

```diff
@@ -0,0 +1,168 @@
+package com.sigma.cinematicketpro.controller;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.sigma.cinematicketpro.TestUtils;
+import com.sigma.cinematicketpro.dto.MovieDto;
+import com.sigma.cinematicketpro.entity.Movie;
+import com.sigma.cinematicketpro.exception.ResourceNotFoundException;
+import com.sigma.cinematicketpro.mapper.MovieMapper;
+import com.sigma.cinematicketpro.service.MovieService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static com.sigma.cinematicketpro.TestUtils.getObjectMapper;
+import static com.sigma.cinematicketpro.TestUtils.mapMvcResultIntoObject;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = MovieController.class)
+@ExtendWith(MockitoExtension.class)
+class MovieControllerTest {
+    protected static final List<Movie> movieList = TestUtils.getValidMovies();
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private MovieService movieService;
+    @MockBean
+    private MovieMapper movieMapper;
+
+    @Test
+    void shouldReturnListOfMovies_WhenGetAllMovies() throws Exception {
+        List<Movie> expectedMovieList = movieList;
+        when(movieService.getAllMovies()).thenReturn(expectedMovieList);
+
+        RequestBuilder request = MockMvcRequestBuilders.get("/movies");
+
+        MvcResult response = mockMvc.perform(request)
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andReturn();
+
+        List<Movie> actualMovieList = mapMvcResultIntoObject(response, new TypeReference<>() {
+        });
+
+        assertThat(expectedMovieList).isEqualTo(actualMovieList);
+        verify(movieService, times(1)).getAllMovies();
+    }
+
+    @Test
+    void shouldThrowResourceNotFoundException_WhenGetMovieByIdAndMovieIdDoesntExist() throws Exception {
+        String movieId = "movieId";
```

Comment:
bad example of id

Reply:
Can you please explain in more detail why this Id is bad? 
As for me it's much more efficient to see "movieId" in logs instead of something like this "507f1f77bcf86cd799439011".